modules/preferences/main/index.coffee

```diff
@@ -286,8 +286,7 @@ class Preferences extends hx.EventEmitter
 
   applyTimezoneOffset: (date, offset) ->
     offset ?= @_.timezoneOffsetLookup(@timezone(), date.getTime()) || 0
-    utc = date.getTime() + (date.getTimezoneOffset() * 60000)
-    new Date(utc + offset * 60 * 60 * 1000)
+    new Date(date.getTime() + -1 * offset * 60 * 60 * 1000)
```

Comment:
I added a positive time-zone. Interestingly, when you change the date to this time-zone manually  - I mean the PC system time the tests not pass. I will debug more this evening after work and will try your suggestion.